pilot/pkg/xds/ep_filters.go

```diff
@@ -134,6 +155,7 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocalityEndpoint
 
 			// Generate the EDS endpoint for this gateway.
 			gwEp := &endpoint.LbEndpoint{
+				HealthStatus: healthStatus,
```

Comment:
We should respect the env

Reply:
I can add a check, but it seems redundant: if SendUnhealthyEndpoints is off, unhealthy endpoints are not on the list and never considered, so the status will be always set to Healthy. Otherwise the check is always required to prevent the healthy gateway for an unhealthy endpoint set.

I can probably optimize by moving the calculation of aggregate health under an env check.

Reply:
I mean we should only mark the HealthStatus only when `SendUnhealthyEndpoints` is on

Reply:
Yes, I understand that. Like I said, it seems redundant, because the default (unset) status is UNKNOWN which is translated to HEALTHY by Envoy, so setting it to HEALTHY explicitly without checking for the SendUnhealthyEndpoints flag effectively changes nothing. On the other hand setting it to UNHEALTHY assumes the knowledge of unhealthy endpoints which is only available when the flag is enabled.